src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3239,14 +3245,42 @@ internal virtual ImmutableArray<LabelSymbol> Labels
                 // (i.e. doesn't work with "params" or optional parameters).
                 if (!method.Parameters.Any())
                 {
-                    candidateMethods.Add((MethodSymbol)member);
+                    candidateMethods.Add(method);
                 }
             }
 
-            MethodSymbol patternMethod = PerformPatternOverloadResolution(patternType, candidateMethods, syntaxExpr, warningsOnly, diagnostics, syntaxTree, messageID);
-
+            patternMethod = PerformPatternOverloadResolution(patternType, candidateMethods, syntaxExpr, warningsOnly, diagnostics, syntaxTree, messageID);
             candidateMethods.Free();
 
+
+            // if we failed to find a method via normal lookup, see if there is an extension method that satisifies it
+            if (patternMethod == null && searchExtensionMethods)
```

Comment:
Use `pattern is null` instead of `== null` if you want to determine if the reference has the value `null`.